src/distributions/integer.rs

```diff
@@ -109,53 +107,60 @@ impl_nzint!(NonZeroU64, NonZeroU64::new);
 impl_nzint!(NonZeroU128, NonZeroU128::new);
 impl_nzint!(NonZeroUsize, NonZeroUsize::new);
 
+
 #[cfg(feature = "simd_support")]
-macro_rules! simd_impl {
-    ($(($intrinsic:ident, $vec:ty),)+) => {$(
-        impl Distribution<$intrinsic> for Standard {
-            #[inline]
-            fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> $intrinsic {
-                $intrinsic::from_bits(rng.gen::<$vec>())
+macro_rules! packed_simd_types_impl {
+    ($($ty:ty),+) => {
+        $(
+            impl Distribution<$ty> for Standard {
+                #[inline]
+                fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> $ty {
+                    let mut vec: $ty = <$ty>::default();
+                    unsafe {
+                        let ptr = &mut vec;
+                        let b_ptr = &mut *(ptr as *mut $ty as *mut [u8; mem::size_of::<$ty>()]);
+                        rng.fill_bytes(b_ptr);
+                    }
+                    vec.to_le()
```

Comment:
I think this is correct, but we should really use `from_bits` like the old code to avoid `unsafe` (but do use `fill_bytes` instead of `gen`).

Unfortunately `from_bits` is not documented on docs.rs; I just dropped [a PR](https://github.com/rust-lang/packed_simd/pull/336) for that.

Reply:
I'm confused by this. Do you mean use `fill_bytes` on a regular array and then `from_slice_unaligned`? That would avoid all unsafe.

Reply:
Hmm, I hadn't figured on `Simd<[u8; 2]>` etc. being hard to construct from an array. Maybe my suggestion doesn't make sense then.

Reply:
We could do something like
```rust
let mut bytes = [0_u8; mem::size_of::<$ty>()];
rng.fill_bytes(&mut bytes);
let vec = $ty::from_bits($u8xN::from_slice_unaligned(&bytes));
vec.to_le()
```
but usizexN don't have `from_bits`,